### ISSUE TITLE
False positive not-callable after destructuring

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#3925]
### Steps to reproduce
```python
class X:
    def __init__(self, f=None):
        self.f1, self.f2 = f or (None, None)

    def g(self, x):
        if self.f1:
            return self.f1(x)
        return x

    def h(self, y):
        return self.f2(y) if self.f2 else y
```

### Current behavior

```
pylint.py:7: [E1102(not-callable), X.g] self.f1 is not callable
pylint.py:11: [E1102(not-callable), X.h] self.f2 is not callable
```

### Expected behavior

No errors

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.8.6 (default, Oct  8 2020, 14:06:32) 
[Clang 12.0.0 (clang-1200.0.32.2)]
```

**Comment by hippo91** [comment]
@sam-s thanks for the report.
IMHO `pylint`, here, is right. From the snippet you gave, i don't see how `pylint` can know that `f1` and/or `f2` are callables.
The only value that can be inferred is that `f` is None and thus `f1` and `f2`.

**Comment by hippo91** [comment]
@sam-s I close the issue, but if you have more information i'll reopen it if needed.

**Comment by sam-s** [comment]
`pylint` is wrong because there is an explicit check before `f1` and `f2` are called.
Moreover, the following code
```python
class Y:
    def __init__(self, f=None):
        self.f = f or None

    def m(self, x):
        if self.f:
            return self.f(x)
        return x
```
produces *no* diagnostics

**Comment by hippo91** [comment]
@sam-s you are right, my interpretation was wrong. 
However i probably found the reason of the difference between `class X` and `class Y`.
I think there is a mistake in the initialization of `f1` and `f2`. IMHO it should be written:
```
self.f1, self.f2 = f, f or (None, None)  # Note the second f
```
Linting the following code:
```
#pylint:disable=missing-module-docstring, missing-class-docstring
#pylint:disable=missing-function-docstring, invalid-name
#pylint:disable=too-few-public-methods
class X:
    def __init__(self, f=None):
        self.f1, self.f2 = f, f or (None, None)

    def g(self, x):
        if self.f1:
            return self.f1(x)
        return x

    def h(self, y):
        return self.f2(y) if self.f2 else y
```
doesn't emit any messages.

**Comment by sam-s** [comment]
There are no errors in the `X` code. The `f` argument should be a `tuple` or functions, not a function.

**Comment by hippo91** [comment]
@sam-s ok i understand. I reopen the issue even if i cannot imagine a way for `pylint` to understand it.

**Comment by sam-s** [comment]
Note that replacing `X.__init__` with
```python
    def __init__(self, f=(None,None)):
        self.f1, self.f2 = f
```
does **not** remove the errors
